src/simplewallet/simplewallet.cpp

```diff
@@ -5446,20 +5446,28 @@ void simple_wallet::on_money_received(uint64_t height, const crypto::hash &txid,
     std::vector<tx_extra_field> tx_extra_fields;
     parse_tx_extra(tx.extra, tx_extra_fields); // failure ok
     tx_extra_nonce extra_nonce;
-    if (find_tx_extra_field_by_type(tx_extra_fields, extra_nonce))
+    tx_extra_pub_key extra_pub_key;
```

Comment:
Looking back at this patch now - why search for the `tx_extra_pub_key` here? I don't recall this field being a requirement for payment id's?